worker-workflow/src/main/resources/workflow-control.js

```diff
@@ -193,6 +193,47 @@ function applyActionDetails(document, actionDetails, terminateOnFailure) {
     }
 }
 
+function onAfterProcessDocument(e){
+    haveFailuresChanged(e.document);
+}
+
+function haveFailuresChanged(document) {
```

Comment:
Change the function name - it sounds like it returns a boolean but actually it processes the failures.